gtt/gtt_display.py

```diff
@@ -182,3 +182,411 @@ def update_bar_value(self, bar_id: IdType, value: int):
+        )
+
+    def clear_all_traces(self):
+        self._conn.write(
+            bytes.fromhex('FE 77')
+        )
+
+    def draw_rectangle(self, x_pos: int, y_pos: int, width: int, height: int):
+        """Creates a rectangle which is really just a single bar.
+        :param x_pos: the distance from the left edge of the screen in pixels
+        :param y_pos: the distance from the top edge of the screen in pixels
+        :param width: the width of the rectangle in pixels
+        :param height: the height of the bar in pixels
+        """
+
+        self._validate_x(x_pos, x_pos + width - 1)
+        self._validate_y(y_pos, y_pos + height - 1)
+        self._conn.write(
+            bytes.fromhex('FE 72') +
+            ints_to_signed_shorts(x_pos, y_pos, width, height)
+        )
+
+    def create_a_label(self, label_id: int, x_pos: int, y_pos: int, width: int, height: int, font: int, R: int, G: int,
+                       B: int, size: int,
+                       value: str, rot: int, VJst: FontAlignVertical = FontAlignVertical.TOP,
+                       HJst: FontAlignHorizontal = FontAlignHorizontal.LEFT):
+        """Creates a label in a portion of the screen
+        :param label_id: index used to identify this label in the label list
+        :param x_pos: the distance from the left edge of the screen in pixels
+        :param y_pos: the distance from the top edge of the screen in pixels
+        :param width: the width of the bar in pixels
+        :param height: the height of the bar in pixels
+        :param rot: the rotation of the text within the label.
+        :param VJst: the vertical justification of text within the label
+        :param HJst: the horizontal justification of text within the label
+        :param font: the Font index of a previously loaded font to be used for the label.
+        :param R: the intensity of red, 0 to 255, used for label font color
+        :param G: the Intensity of green, 0 to 255, used for label font color
+        :param B: the Intensity of blue, 0 to 255, used for label font color
+        :param value: New ASCII formatted string to display within the label. String should be a single line in height.
+        :param size: Size of the font
+        """
+
+        self._validate_x(x_pos, x_pos + width - 1)
+        self._validate_y(y_pos, y_pos + height - 1)
+        label_id = self._resolve_id(label_id, new=True)
+
+        self._conn.write(
+            bytes.fromhex('FE 10') +
+            label_id.to_bytes(1, 'big') +
+            ints_to_signed_shorts(x_pos, y_pos, width, height, rot) +
+            VJst.to_bytes(1, 'big') +
+            HJst.to_bytes(1, 'big') +
+            font.to_bytes(1, 'big') +
+            hex_colors_to_bytes('FFFFFF')
+        )
+
+        self.update_label(label_id, value)
+        self.set_label_font_color(label_id, R, G, B)
+        self.set_label_font_size(label_id, size)
+        self.set_label_background_color(label_id, RB=0xFF, GB=0xFF, BB=0xFF)
```

Comment:
Why aren't these parameters? Look at how the bar handles foreground and background colors.